arrow-select/src/take.rs

```diff
@@ -448,8 +449,10 @@ fn take_bits<I: ArrowPrimitiveType>(
             let mut output_buffer = MutableBuffer::new_null(len);
             let output_slice = output_buffer.as_slice_mut();
             nulls.valid_indices().for_each(|idx| {
-                if values.value(indices.value(idx).as_usize()) {
-                    bit_util::set_bit(output_slice, idx);
+                // SAFETY: idx<indices.len()
+                if values.value(unsafe { indices.value_unchecked(idx).as_usize() }) {
+                    // SAFETY: idx < indices.len()
```

Comment:
I think the safety of this line relies on the fact that `MutableBuffer::new_len()` was created with sufficeint space for indices.len() bits
```suggestion
                    // SAFETY: MutableBuffer was created with space for indices.len() bit, and idx < indices.len()
```